src/mlpack/methods/ann/loss_functions/multilabel_softmargin_loss.hpp

```diff
@@ -0,0 +1,121 @@
+ *
+ * Definition of the Multi Label Soft Margin Loss function.
+ * It is a criterion that optimizes a multi-label one-versus-all loss based on
+ * max-entropy, between input x and target y of size (N, C) where N is the
+ * batch size and C is the number of classes.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_ANN_LOSS_FUNCTION_MULTILABEL_SOFTMARGIN_LOSS_HPP
+#define MLPACK_ANN_LOSS_FUNCTION_MULTILABEL_SOFTMARGIN_LOSS_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class MultiLabelSoftMarginLoss
+{
+ public:
+  /**
+   * Create the MultiLabelSoftMarginLoss object.
+   *
+   * @param weight A manual rescaling weight given to each class. Weight is a
+   *               (1, C) vector where C is the number of classes.
+   * @param num_classes The number of classes, C.
+   * @param reduction Specifies the reduction to apply to the output. If false,
+   *                  'mean' reduction is used, where sum of the output will be
+   *                  divided by the number of elements in the output. If
+   *                  true, 'sum' reduction is used and the output will be
+   *                  summed. It is set to true by default.
+   */
+  MultiLabelSoftMarginLoss(arma::mat weight,
+                           const size_t num_classes,
+                           const bool reduction = true);
+
+  /**
+   * Computes the Multi Label Soft Margin Loss function.
+   *
+   * @param input Input data used for evaluating the specified function.
+   * @param target The target vector with same shape as input.
+   */
+  template<typename InputType, typename TargetType>
+  typename InputType::elem_type Forward(const InputType& input,
+                                        const TargetType& target);
+
+  /**
+   * Ordinary feed backward pass of a neural network.
+   *
+   * @param input The propagated input activation.
+   * @param target The target vector.
+   * @param output The calculated error.
+   */
+  template<typename InputType, typename TargetType, typename OutputType>
+  void Backward(const InputType& input,
+                const TargetType& target,
+                OutputType& output);
+
+  //! Get the input parameter.
+  InputDataType& InputParameter() const { return inputParameter; }
+  //! Modify the input parameter.
+  InputDataType& InputParameter() { return inputParameter; }
```

Comment:
I think I raised this question in another PR, Are these parameters needed?

> Hmm, Could you explain why we need to do this, I might be missing something. I don't think this is needed here. 
Refer : [1](https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/loss_functions/mean_squared_logarithmic_error.hpp), [2](https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/loss_functions/kl_divergence.hpp), [3](https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/loss_functions/mean_squared_error.hpp)

Reply:
Right! I didn't notice your comments in those PRs previously and wasn't aware. Removed them.